app/renderer/js/main.js

```diff
@@ -0,0 +1,251 @@
+		this.isLoading = true;
+		$webView.addEventListener('dom-ready', this.endLoading.bind(this, index));
+		this.registerListeners($webView, index);
+		this.zoomFactors[index] = 1;
+	}
+
+	startLoading(url, index) {
+		const $activeWebView = document.getElementById(`webview-${this.activeTabIndex}`);
+		if ($activeWebView) {
+			$activeWebView.classList.add('disabled');
+		}
+		const $webView = document.getElementById(`webview-${index}`);
+		if ($webView === null) {
+			this.initWebView(url, index, this.settingsTabIndex === index);
+		} else {
+			$webView.classList.remove('disabled');
+		}
+	}
+
+	endLoading(index) {
+		const $webView = document.getElementById(`webview-${index}`);
+		this.isLoading = false;
+		$webView.classList.remove('loading');
+	}
+
+	initActions() {
+		this.$addServerButton.addEventListener('click', this.openSettings.bind(this));
+		this.$settingsButton.addEventListener('click', this.openSettings.bind(this));
+	}
+
+	openSettings() {
+		if (this.settingsTabIndex !== -1) {
+			this.activateTab(this.settingsTabIndex);
+			return;
+		}
+		const url = 'file://' + __dirname + '/preference.html';
+
+		const settingsTabTemplate = `
+				<div class="tab" domain="${url}">
+					<div class="server-tab settings-tab">
+						<i class="material-icons md-48">settings</i>
+					</div>
+				</div>`;
+		this.initTab({
+			alias: 'Settings',
+			url,
+			template: settingsTabTemplate
+		});
+
+		this.settingsTabIndex = this.$tabsContainer.childNodes.length - 1;
+		this.activateTab(this.settingsTabIndex);
+	}
+
+	activateTab(index) {
+		if (this.isLoading) {
+			return;
+		}
+
+		if (this.activeTabIndex !== -1) {
+			if (this.activeTabIndex === index) {
+				return;
+			} else {
+				this.getTabAt(this.activeTabIndex).classList.remove('active');
+			}
+		}
+
+		const $tab = this.getTabAt(index);
+		$tab.classList.add('active');
+
+		const domain = $tab.getAttribute('domain');
+		this.startLoading(domain, index);
+		this.activeTabIndex = index;
+	}
+
+	insertNode(html) {
+		const wrapper = document.createElement('div');
+		wrapper.innerHTML = html;
+		return wrapper.firstElementChild;
+	}
+
+	getTabAt(index) {
+		return this.$tabsContainer.childNodes[index];
+	}
+
+	registerListeners($webView, index) {
+		$webView.addEventListener('new-window', event => {
+			const {url} = event;
+			const domainPrefix = this.domainUtil.getDomain(this.activeTabIndex).url;
+			if (linkIsInternal(domainPrefix, url) && url.match(skipImages) === null) {
+				event.preventDefault();
+				return $webView.loadURL(url);
+			}
+			event.preventDefault();
+			shell.openExternal(url);
+		});
+
+		$webView.addEventListener('page-title-updated', event => {
+			const {title} = event;
+			if (title.indexOf('Zulip') === -1) {
+				return;
+			}
+
+			let messageCount = (/\(([0-9]+)\)/).exec(title);
+			messageCount = messageCount ? Number(messageCount[1]) : 0;
+
+			this.badgeNumberList[index] = messageCount;
+
+			const sum = this.badgeNumberList.reduce((a, b) => {
```

Comment:
This PR shows unread message badge counts as a total for all organizations. The unread messages also sum up as one visits the organizations... 

I.e.: When I open the app, the badge count shows the unread messages from organization 1. When I navigate via the side panel to organization 2, the badge count sums up the unread messages from both organization 1 & 2.. and so on..

I think it would be more desireable to show the unread badge count as either,

1. The sum of all of the users organizations
2. Or the badge count should switch when a user switches organizations.

By 'badge count' I am referring to both the tray and dock numbers on OSX. Have not tried on other OSs.

/cc @akashnimare